transport/grpc/outbound.go

```diff
@@ -94,3 +108,47 @@ func callDownstream(
 
 	return &transport.Response{Body: closer, Headers: headers}, nil
 }
+
+func getErrFromGRPCError(err error, treq *transport.Request, ttl time.Duration) error {
+	switch grpc.Code(err) {
+	// TIMEOUT
+	case codes.DeadlineExceeded:
+		return errors.ClientTimeoutError(treq.Service, treq.Procedure, ttl)
+
+	// BAD REQUEST
+	case codes.Unimplemented:
+		fallthrough
+	case codes.InvalidArgument:
+		fallthrough
+	case codes.NotFound:
+		return errors.RemoteBadRequestError(grpc.ErrorDesc(err))
+
+	// UNEXPECTED
+	case codes.Canceled:
+		fallthrough
+	case codes.AlreadyExists:
+		fallthrough
+	case codes.PermissionDenied:
+		fallthrough
+	case codes.Unauthenticated:
```

Comment:
You can consolidate these case statements instead of using fallthrough
everywhere.

```
case codes.Canceled, codes.AlreadyExists, codes.PermissionDenied,
    codes.Unauthenticated, codes.FailedPrecondition, ...:
    return errors.RemoteUnexpectedError(grpc.ErrorDesc(err))
```